src/uct/sm/mm/base/mm_md.c

```diff
@@ -69,14 +69,15 @@ ucs_status_t uct_mm_seg_new(void *address, size_t length, uct_mm_seg_t **seg_p)
 
 void uct_mm_md_query(uct_md_h md, uct_md_attr_v2_t *md_attr, uint64_t max_alloc)
 {
-    md_attr->flags            = UCT_MD_FLAG_RKEY_PTR | UCT_MD_FLAG_NEED_RKEY;
-    md_attr->max_reg          = 0;
-    md_attr->max_alloc        = 0;
-    md_attr->alloc_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_HOST);
-    md_attr->cache_mem_types  = UCS_BIT(UCS_MEMORY_TYPE_HOST);
-    md_attr->access_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
-    md_attr->detect_mem_types = 0;
-    md_attr->dmabuf_mem_types = 0;
+    md_attr->flags                  = UCT_MD_FLAG_RKEY_PTR | UCT_MD_FLAG_NEED_RKEY;
+    md_attr->max_reg                = 0;
+    md_attr->max_alloc              = 0;
+    md_attr->alloc_mem_types        = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    md_attr->cache_mem_types        = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    md_attr->access_mem_types       = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    md_attr->detect_mem_types       = 0;
+    md_attr->dmabuf_mem_types       = 0;
+    md_attr->reg_nonblock_mem_types = 0;
```

Comment:
why need to initialize this, and not md_attr->reg_mem_types?

Reply:
i think we do not need both, because `md_attr->flags` does not include `UCT_MD_FLAG_REG`

Reply:
right